### PR TITLE
Add --pre-import/pre_import argument

### PR DESCRIPTION
**Pull request by pentschev**
Add argument to pre-import libraries as a Worker plugin to prevent long import times bleeding through later Dask operations.

### CODE REVIEW

File: dask_cuda/cli/dask_cuda_worker.py [hunk 1]
```diff
@@ -277,6 +277,13 @@
     help="""Use a different class than Distributed's default (``distributed.Worker``)
     to spawn ``distributed.Nanny``.""",
 )
+@click.option(
+    "--pre-import",
+    default=None,
```
Comment (randerzander) [inline dask_cuda/cli/dask_cuda_worker.py:282]: Should we not include `cudf` and `rmm` by default?

If not, because we don't want to assume all dask-cuda users are cudf users, this would be a point in favor of thinking more about the concept of dask-cuda "profiles" which include the defaults most users will want without knowing they want them.

Comment (quasiben) [reply dask_cuda/cli/dask_cuda_worker.py:282]: Building a [RAPIDS worker plugin](http://distributed.dask.org/en/stable/plugins.html#worker-plugins) would make sense.  But yes, for now dask-cuda is still general purpose and we shouldn't include it by default.

File: dask_cuda/cli/dask_cuda_worker.py [hunk 2]
```diff
@@ -277,6 +277,13 @@
...
+    help="""Pre-import libraries as a Worker plugin to prevent long import times
+    bleeding through later Dask operations. Should be a list of comma-separated names,
+    such as "cudf,rmm".""",
```
Comment (shwina) [inline dask_cuda/cli/dask_cuda_worker.py:285]: Apologies for the drive-by comment, but wouldn't a `List[str]` be more appropriate here?

File: dask_cuda/cli/dask_cuda_worker.py (hunk 1 above)
Comment (jakirkham) [reply dask_cuda/cli/dask_cuda_worker.py:282]: Yeah we do have people come by periodically who are simply using Dask-CUDA alone. They might have issues if we added other hard requirements like these

That said, maybe we could include an example in the docs?

File: dask_cuda/cli/dask_cuda_worker.py (hunk 2 above)
Comment (jakirkham) [reply dask_cuda/cli/dask_cuda_worker.py:285]: There is [`nargs` support]( https://click.palletsprojects.com/en/8.0.x/options/#multi-value-options ) in Click. Is that what you are referring to?

**Comment by jakirkham** [comment]
Not sure how doable this would be, but would it make sense to add a test?

File: dask_cuda/cli/dask_cuda_worker.py (hunk 2 above)
Comment (shwina) [reply dask_cuda/cli/dask_cuda_worker.py:285]: Ah hmm that wouldn't work either as it's not a fixed number of args. I'm just going to close this suggestion as I don't think there's a better way to do it, honestly.

Comment (jakirkham) [reply dask_cuda/cli/dask_cuda_worker.py:285]: Is it? I think if [`nargs=-1`]( https://click.palletsprojects.com/en/8.0.x/arguments/#variadic-arguments ), we'd get the intended behavior

Comment (pentschev) [reply dask_cuda/cli/dask_cuda_worker.py:285]: `nargs=-1` seems to only apply to variadic arguments though (i.e., "unnamed" arguments), but [not support for options](https://github.com/pallets/click/blob/main/src/click/core.py#L2492) as is the case for `--pre-import`.

Comment (jakirkham) [reply dask_cuda/cli/dask_cuda_worker.py:285]: Makes sense. Then let's leave as-is. There doesn't seem to be a good alternative

File: dask_cuda/cli/dask_cuda_worker.py (hunk 1 above)
Comment (pentschev) [reply dask_cuda/cli/dask_cuda_worker.py:282]: This is the example that is in the argument description itself, see https://github.com/rapidsai/dask-cuda/pull/854/files#diff-b3d47096fc004d0fe86d1630b46845c70e79a9914b065ba2b52c4fc073dd1621R283-R285 and https://github.com/rapidsai/dask-cuda/pull/854/files#diff-245fed94d1f88f78dbd6898226b22c3cb2084eb9e8e7c8994beee67c44461f6aR169-R171 . I'm not sure I see what more is being suggested we do here.

**Comment by pentschev** [comment]
> Not sure how doable this would be, but would it make sense to add a test?

Quite time consuming to get smooth tests for this, but now done in https://github.com/rapidsai/dask-cuda/pull/854/commits/b21f791efa1ba25449c54ed3f751c947e2b47a7a .

**Comment by jakirkham** [comment]
Ouch! Sorry to hear that Peter 😞

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@gpucibot merge

**Comment by jakirkham** [comment]
Thanks Peter! 😄

File: dask_cuda/cli/dask_cuda_worker.py (hunk 1 above)
Comment (jakirkham) [reply dask_cuda/cli/dask_cuda_worker.py:282]: Yeah agree docs are sufficient already then

Maybe @randerzander could share a bit more of what he is looking for with Dask-CUDA profiles in a new issue and we could discuss that? 😉
